### PR TITLE
chore(release): version bump and update changelog

### PR DESCRIPTION
**Pull request by cnishina**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,211 @@
+# 5.0.0
+
+This version includes big changes around upgrading to selenium-webdriver 3.0.x.
+See the [selenium-webdriver changelog](https://github.com/SeleniumHQ/selenium/blob/master/javascript/node/selenium-webdriver/CHANGES.md). Some of the breaking change highlights are summarized below:
+
+- Minimum node version 6.9.x for selenium-webdriver dependency.
+- Jasmine typescript users will need to include the `@types/jasmine`. This
```
Comment (juliemr) [inline CHANGELOG.md]: TypeScript capitalization

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,211 @@
+# 5.0.0
+
+This version includes big changes around upgrading to selenium-webdriver 3.0.x.
+See the [selenium-webdriver changelog](https://github.com/SeleniumHQ/selenium/blob/master/javascript/node/selenium-webdriver/CHANGES.md). Some of the breaking change highlights are summarized below:
+
+- Minimum node version 6.9.x for selenium-webdriver dependency.
```
Comment (juliemr) [inline CHANGELOG.md]: Just "minimum supported node version is now 6.8.x"

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,211 @@
...
+- Jasmine typescript users will need to include the `@types/jasmine`. This
+  is to not cause conflicts with for mocha typings.
+- Removed `protractor.wrapDriver()`.
+- You can no longer use `repl` command from within `browser.pause()`. Instead,
+  use `browser.explore()` to directly enter the `repl`.
+- Methods removed: `WebDriver.prototype.isElementPresent`,
+  `WebElement.prototype.getRawId`, `WebElement.prototype.getInnerHtml`, and
+  `WebElement.prototype.getOuterHtml`.
+- Selenium-webdriver `ErrorCodes` have been deprecated.
```
Comment (juliemr) [inline CHANGELOG.md]: Deprecated or removed?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,211 @@
...
+- When testing with Firefox 47, use `marionette: false` to use legacy Firefox
+  driver.
+
+For the 5.0.0 release, we are still using the selenium standalone server 2.53.1
+and recommend using Firefox 47.
+
+## Features
+
+- ([ec93c4a](https://github.com/angular/protractor/commit/ec93c4ab882991410ad9d3f52d87c0f5ec947641))
+  chore(cli): **breaking change** throw errors on unknown flags (#3921)
```
Comment (juliemr) [inline CHANGELOG.md]: Maybe clarify 'unknown CLI flags'?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,211 @@
...
+- Adding cookies have been changed from
+  `browser.manage().addCookie('testcookie', 'Jane-1234');` to
+  `browser.manage().addCookie({name:'testcookie', value: 'Jane-1234'});`
+- When testing with Firefox 47, use `marionette: false` to use legacy Firefox
+  driver.
+
+For the 5.0.0 release, we are still using the selenium standalone server 2.53.1
+and recommend using Firefox 47.
+
+## Features
```
Comment (juliemr) [inline CHANGELOG.md]: We should have a `## Breaking Changes` section like we've done in the past. You can include everything that's a breaking change from features/deps here (and not include it in the other categories, or include just the title, I don't care)

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,211 @@
...
+  feat(rootEl): ***breaking change*** auto-detect the root element better (#3928)
+
+  This is a breaking change because it changes the default root element behavior
+ and removes the `config.useAllAngular2AppRoots` flag.  Modern angular apps now
+ default to using all app hooks, and ng1 apps now check several places, notably
+ the element the app bootstraps to.
+
+  closes #1742
+- ([604fdbf](https://github.com/angular/protractor/commit/604fdbf064cc2785a2e745747beeaeb45d27f8ed))
+  cleanup(config): **breaking change** Remove redundant sauceProxy config (#3868)
```
Comment (juliemr) [inline CHANGELOG.md]: Why is this a breaking change? The commit message doesn't list it as such.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,211 @@
...
+
+  BREAKING CHANGE
+
+  You can no longer use the `repl` command from within `browser.pause()`. Instead,
+ use `browser.explore()` to directly enter the repl.
+- ([8196059](https://github.com/angular/protractor/commit/819605933d2dfef70b4332a727b3b3830e306817))
+  chore(dependency): switch to webdriver-manager 11.1.0 and remove
+  `--versions.chrome 2.26` from circle.yml (#3865)
+- ([397bf65](https://github.com/angular/protractor/commit/397bf65e088b640cf3612f9da678180f49939b84))
+  deps(update): move @types/jasmine to devDependencies (#3795)
```
Comment (juliemr) [inline CHANGELOG.md]: This one should get listed as a breaking change, and have the solution (add @types to your deps) clearly listed.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,211 @@
...
+  Note: with the upgrade, there might have breaking changes. Please see the
+  selenium-webdriver changelog.
+
+  - See [selenium-webdriver changelog](https://github.com/SeleniumHQ/selenium/blob/master/javascript/node/selenium-webdriver/CHANGES.md)
+
+    - Removed method `WebDriver.prototype.isElementPresent`
+    - Removed method `WebElement.prototype.getRawId`
+    - Removed `getInnerHtml` and `getOutterHtml`
+
+    - Dependency required for upgrade: use `jasminewd2@0.1.0-beta.0`.
```
Comment (juliemr) [inline CHANGELOG.md]: This is now 0.1.0

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,211 @@
...
+      required to pass "marionette: false" in the capabilities. If you do not pass
+      marionette to false, it will look for gecko driver in the PATH.
+    - Added a TODO to support FF after 48+ with direct connect and gecko driver.
+
+    - Updated `browser.manage().addCookie('testcookie', 'Jane-1234');` to use
+      `browser.manage().addCookie({name:'testcookie', value: 'Jane-1234'});`
+
+    - Updated debug commons for breakpoint updated to selenium-webdriver
+      `lib/http` line 432.
+    - Debugger currently does not work. So `elementExplorer` for this current
```
Comment (juliemr) [inline CHANGELOG.md]: Remove this line now :)

File: CHANGELOG.md (hunk 5 above)
Comment (juliemr) [inline CHANGELOG.md]: Some of the stuff from the quick list above should probably be formalized and moved to the breaking changes section.

File: CHANGELOG.md (hunk 6 above)
Comment (mgiambalvo) [inline CHANGELOG.md]: It removes the `sauceProxy` config field, and uses `webDriverProxy` when creating the SauceLabs client.

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,211 @@
...
+- ([397bf65](https://github.com/angular/protractor/commit/397bf65e088b640cf3612f9da678180f49939b84))
+  deps(update): move @types/jasmine to devDependencies (#3795)
+
+  - update outdated dependencies
+  - move @types/jasmine to devDependencies
+   closes #3792
+- ([a3e8b43](https://github.com/angular/protractor/commit/a3e8b4319d3e8b049e55e5c3c64a7fdb5a132ddf))
+  deps(selenium-webdriver): upgrade to selenium 3 (#3781)
+
+  Note: with the upgrade, there might have breaking changes. Please see the
```
Comment (mgiambalvo) [inline CHANGELOG.md]: 'there might have been'. Might be redundant since we do list methods that got removed.

File: CHANGELOG.md (hunk 7 above)
Comment (cnishina) [inline CHANGELOG.md]: done.

File: CHANGELOG.md (hunk 10 above)
Comment (cnishina) [inline CHANGELOG.md]: cleaned up.

File: CHANGELOG.md (hunk 9 above)
Comment (cnishina) [inline CHANGELOG.md]: removed debugger / disabled tests comment

File: CHANGELOG.md (hunk 8 above)
Comment (cnishina) [inline CHANGELOG.md]: done.

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,229 @@
...
+See the [selenium-webdriver changelog](https://github.com/SeleniumHQ/selenium/blob/master/javascript/node/selenium-webdriver/CHANGES.md).
+
+For the 5.0.0 release, we are still using the selenium standalone server 2.53.1
+and recommend using Firefox 47. Firefox 48+ currently is not supported.
+
+## Breaking Changes
+
+- Minimum node version is now 6.9.x.
+- When testing with Firefox 47, use the capability option `marionette: false`
+  to use the legacy Firefox driver.
```
Comment (juliemr) [inline CHANGELOG.md]: Let's add an explicit:

After:

```
capabilities: {
  browserName: 'firefox',
  marionette: false
}
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,229 @@
...
+- We moved `@types/jasmine` to a devDependency. This means that Jasmine
+  TypeScript users will need to include the `@types/jasmine` as a project
+  dependency. This is to avoid conflicts for users that prefer mocha typings.
```
Comment (juliemr) [inline CHANGELOG.md]: After:

```
dependencies: {
  '@types/jasmine': '^2.4.0'
}
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,3 +1,229 @@
...
+- Selenium-webdriver methods removed: `WebDriver.prototype.isElementPresent`,
+  `WebElement.prototype.getRawId`, `WebElement.prototype.getInnerHtml`, and
+  `WebElement.prototype.getOuterHtml`.
```
Comment (juliemr) [inline CHANGELOG.md]: Before:

```js
let i = element(locator).getInnerHtml();
```

After:

```js
let i = browser..executeScript("return arguments[0].innerHTML;", element(locator));
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,3 +1,229 @@
...
+- Sending flags that are not recognized by the CLI throws an error. Since flags
+  are a subset of all configuration options, these errors can be silenced with
+  `--disableChecks`.
+- Auto-detection of the root element. This is a breaking change because it
+  changes the default root element behavior and removes the
+  `config.useAllAngular2AppRoots` flag.  Modern angular apps now
+  default to using all app hooks, and ng1 apps now check several places, notably
+  the element the app bootstraps to.
+- `sauceProxy` configuration field has been removed. Use `webDriverProxy`
+  instead.
```
Comment (juliemr) [inline CHANGELOG.md]: Add before/after here too.

**Comment by juliemr** [comment]
LGTM.

**Pull request merged**